### ISSUE TITLE
Offer Tekton task for Quarkus build (including native)

### ISSUE THREAD

**Issue by emmanuelbernard** [quarkusio/quarkus#9338]

**Comment by emmanuelbernard** [comment]
CC @maxandersen

**Comment by maxandersen** [comment]
you mean document an example task for quarkus or something more ?

**Comment by maxandersen** [comment]
related to #6147

**Comment by emmanuelbernard** [comment]
@tqvarnst ^

**Comment by quintesse** [comment]
My first idea was to use [Quarkus S2I builders](https://github.com/quarkusio/quarkus-images) together with the [Tekton S2I Task](https://github.com/tektoncd/catalog/tree/master/task/s2i/0.1) but unfortunately two issues prevent that from working right now:

 1. The Quarkus S2I images use non-standard paths. Now this is certainly allowed by the spec but no other images do this. I opened [an issue to update the paths](https://github.com/quarkusio/quarkus-images/issues/85).
 2. The Tekton S2I task doesn't support non-standard paths. Given the fact it simply uses `s2i` I'm not sure yet what the problem is there. Still investigating.

**Comment by maxandersen** [comment]
@quintesse shouldn't it be more straightforward to just run mvn build directly - skipping the whole s2i completely ? We can't use s2i for native image anyway.

**Comment by quintesse** [comment]
> shouldn't it be more straightforward to just run mvn build directly

Well it seemed more straightforward to use a task that already existed for a builder image especially made for Quarkus

> We can't use s2i for native image anyway.

Why not? There's a special builder image for native builds

**Comment by quintesse** [comment]
> The Tekton S2I task doesn't support non-standard paths

Ok, it's the option for generating the Dockerfile. The task doesn't actually let S2I the building, it just asks S2I to generate a Dockerfile and then delegates building to `buildah`. It's that generated Dockerfile that has the wrong paths.

**Comment by lkrzyzanek** [comment]
Another approach than S2I is just maven + `quarkus-container-image-jib` extension. It's very lightweight because it depends purely on just maven.  See comment in related thread: https://github.com/quarkusio/quarkus/issues/6147#issuecomment-693382489

**Comment by quintesse** [comment]
@lkrzyzanek yes, that's the same path I'm currently taking: https://github.com/quintesse/quarkus-build-task/blob/master/quarkus-build.yml#L61-L72

**Comment by lkrzyzanek** [comment]
@quintesse yep looks nice. I like it.
For Maven caching I have good experience with volume resp. podTemplate rather than storing it in workspace:
example: 
https://github.com/lkrzyzanek/quarkus-tekton-pipeline-example/blob/maven-cache/pipeline/task-maven-with-cache.yaml#L151
and pipelinerun: https://github.com/lkrzyzanek/quarkus-tekton-pipeline-example/blob/maven-cache/pipeline/run/pipelinerun-pvc-cache.yaml#L37

resp. more info in this issue: https://github.com/tektoncd/catalog/issues/514

**Comment by quintesse** [comment]
@lkrzyzanek well it doesn't work 100% yet 😅 

The issue I have with the volumes+podTempalte is that it makes _using_ the Task more complex than necessary. And I already find Tekton inching towards being overly complex. Given the fact that this is supposed to be the official Quarkus Task I'm trying to make it as easy as possible to use. Still, I'll take a good look at your links. Thanks!

**Comment by lkrzyzanek** [comment]
Yeah Tekton is sometimes too complex. I tried multiple workspaces as well and the issue is that it complained when using multiple PVC for each of them. Another disadvantage is that when you don't want to use the cache workspace you have to define it in taskrun/pipeline anyway. With volumes you don't have to. So every design has pros and cons as usual :-)
